include/git2/sys/transport.h

```diff
@@ -286,58 +294,81 @@ typedef enum {
 typedef struct git_smart_subtransport git_smart_subtransport;
 typedef struct git_smart_subtransport_stream git_smart_subtransport_stream;
 
-/* A stream used by the smart transport to read and write data
+/**
+ * A stream used by the smart transport to read and write data
  * from a subtransport */
 struct git_smart_subtransport_stream {
-	/* The owning subtransport */
+	/** The owning subtransport */
 	git_smart_subtransport *subtransport;
 
+	/** Read available data from the stream */
```

Comment:
For those functions we should definitely document their semantics, which aren't immediately obvious (see e.g. #4953). Most importantly, we should specify whether one needs to loop around read/write

Reply:
I'll keep that open while it's being specified then 😉.

Reply:
I've just checked again to make sure I get it right. This is the state with #4953:

- Both `git_proto_stream_write` and `http_stream_write_single` use `git_stream__write_full` and thus no looping is required. We should specify that a substream implementation shall either write all bytes or return an error code.
- `git_proto_stream_read` tries to read `buf_size` many bytes, but may return less bytes
- `http_stream_read` loops (I think) until a complete HTTP response has been parsed. 

So I guess for the reading part we need to specify only that the implementation may read less bytes than requested and thus may require looping. But one can already see that different implementations behave different, too

Reply:
Would you be fine with merging this as it is, and revisit the doc after #4953 has landed and semantics are clear ?